src/explore-education-statistics-admin/src/pages/release/data/components/DataFilesTable.tsx

```diff
@@ -0,0 +1,239 @@
+} from '@admin/routes/releaseRoutes';
+import {
+  DataFile,
+  DataFileImportStatus,
+} from '@admin/services/releaseDataFileService';
+import ButtonGroup from '@common/components/ButtonGroup';
+import ButtonText from '@common/components/ButtonText';
+import Modal from '@common/components/Modal';
+import ReorderableList from '@common/components/ReorderableList';
+import reorder from '@common/utils/reorder';
+import React, { useEffect, useState } from 'react';
+import { generatePath } from 'react-router';
+import styles from './DataFilesTable.module.scss';
+
+interface Props {
+  dataFiles: DataFile[];
+  publicationId: string;
+  releaseId: string;
+  canUpdateRelease?: boolean;
+  isReordering: boolean;
+  onCancelReordering: () => void;
+  onConfirmReordering: (nextSeries: DataFile[]) => void;
+  onDeleteFile: (dataFile: DataFile) => Promise<void>;
+  onStatusChange: (
+    dataFile: DataFile,
+    { totalRows, status }: DataFileImportStatus,
+  ) => Promise<void>;
+}
+
+const DataFilesTable = ({
+  dataFiles: initialDataFiles,
+  publicationId,
+  releaseId,
+  canUpdateRelease,
+  isReordering,
+  onCancelReordering,
+  onConfirmReordering,
+  onDeleteFile,
+  onStatusChange,
+}: Props) => {
+  const [dataFiles, setDataFiles] = useState(initialDataFiles);
+
+  useEffect(() => {
+    setDataFiles(initialDataFiles);
+  }, [initialDataFiles]);
+
+  if (isReordering) {
+    return (
+      <ReorderableList
+        heading="Reorder data files"
+        id="dataFiles"
+        list={dataFiles.map(({ id, title }) => ({
+          id,
+          label: title,
+        }))}
+        onCancel={() => {
+          setDataFiles(initialDataFiles);
+          onCancelReordering();
+        }}
+        onConfirm={() => onConfirmReordering(dataFiles)}
+        onMoveItem={({ prevIndex, nextIndex }) => {
+          const reordered = reorder(dataFiles, prevIndex, nextIndex);
+          setDataFiles(reordered);
+        }}
+        onReverse={() => {
+          setDataFiles(dataFiles.toReversed());
+        }}
+      />
+    );
+  }
+
+  return (
+    <table className={styles.table} data-testid="Data files table">
+      <thead>
+        <tr>
+          <th scope="col">Subject title</th>
+          <th scope="col">Size</th>
+          <th scope="col">Status</th>
+          <th scope="col">Actions</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {dataFiles.map(dataFile => (
+          <tr key={dataFile.title}>
+            <td data-testid="Subject title">{dataFile.title}</td>
+            <td data-testid="Data file size" className={styles.fileSize}>
+              {dataFile.fileSize.size.toLocaleString()} {dataFile.fileSize.unit}
+            </td>
+            <td data-testid="Status">
+              <ImporterStatus
+                className={styles.fileStatus}
+                releaseId={releaseId}
+                dataFile={dataFile}
+                onStatusChange={onStatusChange}
+              />
+            </td>
```

Comment:
Currently looks a bit horrible when a data replacement is in progress as we get a horrible big blue box. 

Think we could get rid of the 'Data' part of the replacement label so we don't get wrapping over 3 lines:

![image](https://github.com/user-attachments/assets/86a900b2-75c2-41ee-9326-fb58359d6e80)